### PR TITLE
Feature Flag Telemetry Support

### PR DESCRIPTION
**Pull request by zhiyuanliang-ms**
## Why this PR?

Populate feature flag id/feature flag reference/etag in feature flag's telemetry.metadata to facilitate Split experimentation.

Follows [PR](https://github.com/Azure/azure-sdk-for-python/pull/35254) and [PR](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/517)

### CODE REVIEW

File: src/AzureAppConfigurationImpl.ts [hunk 1]
```diff
@@ -532,6 +537,83 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         }
         return response;
     }
+
+    async #parseFeatureFlag(setting: ConfigurationSetting<string>): Promise<any> {
```
Comment (linglingye001) [inline src/AzureAppConfigurationImpl.ts:541]: why async, same for line 561

File: src/AzureAppConfigurationImpl.ts [hunk 2]
```diff
@@ -532,6 +537,83 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
...
+        }
+        // In Node.js, use the crypto module's hash function
+        else {
+            const hash = crypto.createHash("sha256").update(data).digest();
+            return hash.toString("base64url");
+        }
+    }
+
+    #createFeatureFlagReference(setting: ConfigurationSetting<string>): string {
+        let featureFlagReference = `${this.#clientEndpoint}kv/${setting.key}`;
```
Comment (linglingye001) [inline src/AzureAppConfigurationImpl.ts:611]: Is there any requirement to featureFlagReference? what if in failover scenario when current endpoint from replica?

Comment (linglingye001) [inline src/AzureAppConfigurationImpl.ts:611]: what if the client endpoint is undefined or CDN endpoints in the future?

File: src/AzureAppConfigurationImpl.ts (hunk 1 above)
Comment (zhiyuanliang-ms) [reply src/AzureAppConfigurationImpl.ts:541]: Line 597

https://stackoverflow.com/questions/57626477/using-javascript-crypto-subtle-in-synchronous-function

File: src/AzureAppConfigurationImpl.ts (hunk 2 above)
Comment (zhiyuanliang-ms) [reply src/AzureAppConfigurationImpl.ts:611]: Please read these two PRs: [.NET provider](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/517) and [python provider](https://github.com/Azure/azure-sdk-for-python/pull/35254)

Comment (zhiyuanliang-ms) [reply src/AzureAppConfigurationImpl.ts:611]: > what if in failover scenario when current endpoint from replica

Then we will use the replica url

Comment (zhiyuanliang-ms) [reply src/AzureAppConfigurationImpl.ts:611]: Then, it will be the CDN endpoint.

File: examples/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "dependencies": {
-    "@azure/app-configuration-provider": "latest",
+    "@azure/app-configuration-provider": "../",
```
Comment (zhiyuanliang-ms) [inline examples/package.json:3]: revert this change

File: examples/helloworld.mjs [hunk 4]
```diff
@@ -19,8 +19,16 @@ const settings = await load(connectionString, {
     selectors: [{
         keyFilter: "app.settings.*"
     }],
-    trimKeyPrefixes: ["app.settings."]
+    trimKeyPrefixes: ["app.settings."],
```
Comment (zhiyuanliang-ms) [inline examples/helloworld.mjs:22]: revert this change

File: src/AzureAppConfigurationImpl.ts (hunk 2 above)
Comment (linglingye001) [inline src/AzureAppConfigurationImpl.ts:611]: Does the casing of "endpoint" affect subsequent hash calculation?

**Review by linglingye001 — APPROVED** [review]

**Review by rossgrambo — APPROVED** [review]
